library/core/src/slice/iter.rs

```diff
@@ -710,18 +710,17 @@ where
             return None;
         }
 
-        let idx_opt = {
-            // work around borrowck limitations
-            let pred = &mut self.pred;
-            self.v.iter().position(|x| (*pred)(x))
-        };
-        match idx_opt {
+        match self.v.iter().position(|x| (self.pred)(x)) {
             None => self.finish(),
             Some(idx) => {
                 let tmp = mem::replace(&mut self.v, &mut []);
```

Comment:
Little nicer?
```suggestion
                let tmp = mem::take(&mut self.v);
```

Reply:
Hunh, I didn't notice that `mem::take` is relatively new. That would explain why this replace-with-default pattern is so common.